h/api/test/views_test.py

```diff
@@ -99,6 +101,34 @@ def test_create(mock_create_annotation, user):
     _assert_event_published('create')
 
 
+@patch("h.api.views.Annotation.save")
+@patch("h.api.views.nipsa.has_nipsa")
+@patch("h.api.views.get_user")
+def test_create_adds_nipsa_flag(get_user, has_nipsa, _):
+    get_user.return_value = Mock(
+        id="test_id", consumer=Mock(key="test_key"))
+    has_nipsa.return_value = True
+    request = DummyRequest(json_body=_new_annotation)
+
+    annotation = views.create(request)
+
+    assert annotation.get("nipsa") is True
+
+
+@patch("h.api.views.Annotation.save")
+@patch("h.api.views.nipsa.has_nipsa")
+@patch("h.api.views.get_user")
+def test_create_does_not_add_nipsa_flag(get_user, has_nipsa, _):
+    get_user.return_value = Mock(
+        id="test_id", consumer=Mock(key="test_key"))
+    has_nipsa.return_value = False
+    request = DummyRequest(json_body=_new_annotation)
+
+    annotation = views.create(request)
+
+    assert not annotation.get("nipsa")
```

Comment:
Another example of where this should be an explicit `is None` check, in my view. This test will pass if the nipsa field is set to the empty string, but that probably won't play well with the elasticsearch filters elsewhere in the PR.